scripts/documentation/builders.ts

```diff
@@ -0,0 +1,97 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { parseJsonSchemaToOptions } from '@angular/cli/utilities/json-schema';
+import { dedent } from 'tslint/lib/utils';
+import { Schematic } from '@angular-devkit/schematics/collection-schema';
+import { CoreSchemaRegistry } from '@angular-devkit/core/src/json/schema';
+import {
+  htmlSelectorFormat,
+  pathFormat
+} from '@angular-devkit/schematics/src/formats';
+
+/**
+ * @WhatItDoes: Generates default documentation from the builders' schema.
+ *    We need to construct an Array of objects containing all the information
+ *    of the builders and their associates schema info. It should be easily
+ *    parsable in order to be used in a rendering process using template. This
+ *    in order to generate a markdown file for each available schematic.
+ */
+const buildersSourceDirectory = path.join(
+  path.resolve(__dirname),
+  '../../build/packages/builders/src'
+);
+const buildersOutputDirectory = path.join(
+  path.resolve(__dirname),
+  '../../docs/builders'
+);
+const builderCollectionFile = path.join(
+  buildersSourceDirectory,
+  'builders.json'
+);
+const builderCollection = fs.readJsonSync(builderCollectionFile).builders;
+const registry = new CoreSchemaRegistry();
+registry.addFormat(pathFormat);
+registry.addFormat(htmlSelectorFormat);
+
+function generateSchematicList(
+  builderCollection: Schematic,
+  registry: CoreSchemaRegistry
+): Promise<Schematic>[] {
+  return Object.keys(builderCollection).map(builderName => {
+    const builder = {
+      name: builderName,
+      ...builderCollection[builderName],
+      rawSchema: fs.readJsonSync(
+        path.join(
+          buildersSourceDirectory,
+          builderCollection[builderName]['schema']
+        )
+      )
+    };
+
+    return parseJsonSchemaToOptions(registry, builder.rawSchema)
+      .then(options => ({ ...builder, options }))
+      .catch(error =>
+        console.error(`Can't parse schema option of ${builder.name}:\n${error}`)
+      );
+  });
+}
+
+function generateTemplate(builder): { name: string; template: string } {
+  let template = dedent`
```

Comment:
const?

Reply:
I can't use `const` because the template variable is modified, enhanced for each options, see above line 71.